drivers/sensor/lsm6dsr/CMakeLists.txt

```diff
@@ -0,0 +1,9 @@
+# Copyright 2023, Andrei Hutanu
+#
+# SPDX-License-Identifier: Apache-2.0
+
+zephyr_library()
+
+zephyr_library_sources(lsm6dsr.c)
+zephyr_library_sources(lsm6dsr_i2c.c)
+zephyr_library_sources_ifdef(CONFIG_LSM6DSR_TRIGGER    lsm6dsr_trigger.c)
```

Comment:
Support to new ST sensors should use STMEMSC HAL. Please look here: https://github.com/zephyrproject-rtos/zephyr/blob/main/drivers/sensor/lsm6dso/CMakeLists.txt#L13